Mu2eG4/geomMacros/clean_volume_names.C

```diff
@@ -0,0 +1,18 @@
+//
+// The volume names in the CSV can contain memory address (i.e. 0xAAAAAAA).
+// For comparing geometries these need to be removed
+//
+
+std::string clean_volume_names(std::string name) {
+  bool all_found = false;
```

Comment:
I like this idea but there is a hook.  We have some distinct physical volumes with identical names.  This is why gdml decorates names with the memory address of the objects.  The right answer is for us to eliminate identical names and add a check when we close the geoemetry.  I will create an issue.  Until this is fixed it's possible that people could get confused if you strip off the trailing memory addresses.

Does it make sense to warn about this?

Reply:
Hi Rob - do you have a specific example to hand? The cleaning is only done when outputting to csv so is not the default, so I think it will be OK. However, in 7e06db41f I've added a warning to the README about using that option

Reply:
Thanks for addding the comments

There  multiple instances of physical volumes named caloPipePV* for *=1,2,3,4,5,6,7,8,9,10. PanelEBKey, StrawPanelEnvelope.

Internally G4 does not use the volume names.  It estabishes identity with memory addresses.  The names are for people.   So it's possible to have a working G4 and ambiguous human readable identities. 

The one operational place that this is bad is the following.  If you ask the volume store to find a volume by name, it gives the first match and ignores the others.  Internally G4 never does this.  Some of our code does - so we need to loop over all volumes and match names to be sure we get them all.  We only do this at startup so the cost is small.

I think it's Ok to keep your code the way it is - few people will be interested in the volumes that need to be distinguished by their addresses.